Increase touch area of close button in settings dialog

BalanaguYashwanth opened the pull request:
Resolved bug by increasing touch area of close button in settings dialog

<!-- Thanks for submitting a PR! Please ensure the following requirements are met in order for us to review your PR -->

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Notes: Increased touch area and padding for close button under the settings dialog and maintaining well in terms of design and colour which suitable exactly for the settings dialog window. 

Fixes https://github.com/vector-im/element-web/issues/22915
Closes https://github.com/matrix-org/matrix-react-sdk/pull/9202

Type: [enhancement/defect/task]

Before - 
 
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/36238382/218301823-d4a8dce0-76dc-4925-8817-b78b2eb83ee5.png">

After -
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/36238382/218301522-f047290e-cde9-4888-b2b7-32769a370ffc.png">

<ins><b>Fixes</b></ins>
Increased touch area and padding for close button under the settings dialog and maintaining well in terms of design and colour which suitable exactly for the settings dialog window. It fixes https Contributed by @BalanaguYashwanth




<!-- CHANGELOG_PREVIEW_START -->
---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Increased touch area and padding for close button under the settings dialog and maintaining well in terms of design and colour which suitable exactly for the settings dialog window. ([\#10145](https://github.com/matrix-org/matrix-react-sdk/pull/10145)). Fixes vector-im/element-web#22915. Contributed by @BalanaguYashwanth.<!-- CHANGELOG_PREVIEW_END -->

gsouquet reviewed (changes requested):
We're adding a wrapper HTML element for purely stylistic reasons, and that is CSS roles.

I believe we should remove the wrapper and virtually increase the touch zone using a pseudo element

```
button::after {
  position: absolute;
  inset: -10px;
  content: "";
}
```

Something of the sort will probably do the trick, but I have not tested that locally.

BalanaguYashwanth commented:
Hello @gsouquet,

Thanks for suggestion, I will check on it.

BalanaguYashwanth commented:
Hey @gsouquet,

I added to css what code you suggested but touch area is not working as expected, can you check from your side once.

Mainly i see CSS and HTML code for that close button is not structured well then i changed it. You can check demo in local of my PR you will definitely see increased touch area and aligned CSS along with HTML in UI

BalanaguYashwanth commented:
Hello,

Anyone can help me out because @gsouquet, suggested some changes but when i tried those are not working on UI, If anyone can guide me in some direction, so that i can improve my PR but existing PR you won't see any problems with it.

But still if you need to improve let me know...

BalanaguYashwanth commented:
Hey @gsouquet ,

Hope you are doing well, Can i expect any feedback.

BalanaguYashwanth commented:
hey thanks for your reply, i am working on to match your solution because unlike other elements to add touch area, this element behaviour in different manner when applying css that you suggested solution, i didn't stoped yet. 

 Working on it, soon will raise with PR with updated code, Anything you want to update me, You can update...

BalanaguYashwanth commented:
Hello @gsouquet,

Updated the css code please look once for increase touch area for cancel button

BalanaguYashwanth commented:
Hello @gsouquet,

I updated based on your inputs, If you review then it will be great

gsouquet commented on res/css/_common.pcss:
```diff
@@ -471,11 +471,10 @@ legend {
+    background-repeat: no-repeat;
+    background-position: center;
+    background-size: cover;
     cursor: pointer;
     position: unset;
```
I don't understand this change. Why is it needed, and I don't think the SVG will be tinted when changing the theme to dark or light.

Could you please revert this?

BalanaguYashwanth replied:
**Background color** also won't tined with dark or light theme because It looks same for two themes, Manipulating the background color and using as cancel icon which is not recommend

Instead using Cancel button svg which is easily maintainable.

germain-gg replied:
This does not need to be changed, so i would request to either revert this as part of this pull request or doing it as described in https://github.com/matrix-org/matrix-react-sdk/blob/develop/docs/icons.md

gsouquet commented on res/css/_common.pcss:
```diff
@@ -484,13 +483,24 @@ legend {
     right: 0;
 }
 
+.mx_Dialog_cancelButton::before{
```
```suggestion
.mx_Dialog_cancelButton::before {
```

gsouquet commented on res/css/_common.pcss:
```diff
@@ -484,13 +483,24 @@ legend {
+.mx_Dialog_cancelButton::before{
+    content: "";
+    position: absolute;
+    height: 50px;
```
Ideally we should avoid hardcoding the height but use `inset` or a combination of `top` and `bottom` so that the height is inferred from the parent element.

More future proof.

BalanaguYashwanth replied:
As per your suggestion, Using top and bottom added the size for the tocuh area. so you can check below area of touch area for this cancel button

<img width="74" alt="image" src="https://github.com/matrix-org/matrix-react-sdk/assets/36238382/9a10ba9e-de75-4cde-ad83-99755a49f861">

dbkr commented:
I'm going to close this as it hasn't had any activity. Feel free to re-open if you (or anyone) wants to keep it moving forward.
